frontend/src/common/components/DropboxChooser/index.js

```diff
@@ -46,17 +54,30 @@ export default class DropboxChooser extends React.Component {
         super(props);
 
         this.onChoose = this.onChoose.bind(this);
+        this.retry = 0;
     }
 
     componentDidMount() {
-        if (!this.isDropboxReady() && !scriptLoadingStarted) {
-            scriptLoadingStarted = true;
-            loadScript(DROPBOX_SDK_URL, {
-                attrs: {
-                    id: SCRIPT_ID,
-                    'data-app-key': this.props.appKey,
-                },
-            });
+        this.loadDropboxApi();
+    }
+
+    onApiLoad = () => {
+        // make sure api is loaded,
+        // called by loadScript after loading is success or failure
+        // if the api is not loaded, try to load in retry delay
+        if (!this.isDropboxReady()) {
+            if (this.retry <= this.props.retryLimit) {
+                setTimeout(() => {
+                    this.retry += 1;
+                    scriptLoadingStarted = false;
+                    this.loadDropboxApi();
```

Comment:
This looks like a recursive call.
I think you should get the timeoutId and then clearTimeout on componentWillUnmount


And looks like the same thing should be done with google drive

Reply:
done https://github.com/eoglethorpe/deeper/pull/228/commits/15d0ca5010d657cb213628cc10e65ca4134d0d60